scrapy/core/scheduler.py

```diff
@@ -321,6 +346,14 @@ def _dq(self):
                         {'queuesize': len(q)}, extra={'spider': self.spider})
         return q
 
+    def _dpq(self):
+        """ Create a new delayed requests priority queue instance, with in-memory storage """
```

Comment:
After a private discussion with @kmike, @wRAR and @VMRuiz, it seemed like the main advantage would be the flexibility. Some users may want request priority to be taken into account regardless of whether the request has been delayed or not, and scheduling into the downloader after a delay, bypassing the scheduler, removes that option.

On the other hand, the scheduler approach makes all behaviors achievable through parameters (scheduler queue settings and request priority values).

Reply:
> For example, currently float request_delay doesn't seem to be prohibited by the docs, but it may be harder to make disk queues work with float priorities, as they're not supported now.

I do not see it as a big issue, since disk queues could pick some granularity (e.g. 0.1s) and adjust the delay accordingly.

But I am OK with going for millisecond granularity on the setting and go for int.

Reply:
> I do not see it as a big issue, since disk queues could pick some granularity (e.g. 0.1s) and adjust the delay accordingly.

With timestamp-based priorities we'll have a lot of disk queues, with 1 element each; we'd need to benchmark it, to see how it works in practice. Current queue implementation doesn't seem right for float priorities, on a first sight.

Reply:
I don't like timestamp-based priorities with the current implementation either. 
The code would need to be refactored to use a single queue.

Reply:
Is disk support for delayed requests something we should implement as part of this pull request, or is that something that can come later?

I do realize that if we change the behavior later to behave like with priority queues, where disk is tried first and memory is only used if serialization fails, it may significantly affect request order, lowering the priority of delayed requests below any unserializable request. But I am not sure how bad that would be, it does not seem too bad to me.